walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,188 @@
 
 package com.google.sps;
 
+import com.google.sps.TimeRange;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  public static final int WHOLE_DAY = 24 * 60;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> meetings = new ArrayList<>();
+    Collection<String> requestedAttendees = request.getAttendees();
+    Collection<String> requestedOptionalAttendees = 
+      request.getOptionalAttendees();
+    long durationOfRequest = request.getDuration();
+
+    // Case 0: if duration too long (longer than a day), return no options
+    if (durationOfRequest > WHOLE_DAY) {
+      return meetings;
+    }
+
+    /** Case 1: GENERAL CASE >> remove conflicting times(from @events) 
+      * from possible meetings
+      */
+    meetings.add(TimeRange.fromStartDuration(0, WHOLE_DAY));
+    ArrayList<Event> optionalOverlappedEvents = new ArrayList<>();
+    for (Event event : events) {
+      Set<String> tempOverlappedEventAttendees = event.getAttendees();
+
+      Set<String> overlappedEventAttendees = new HashSet<String>();
+      Set<String> optionalOverlappedAttendees = new HashSet<String>();
+
+      // Copy attendees into a modifiable collection
+      overlappedEventAttendees = 
+        copyToModifiableCollection(
+          overlappedEventAttendees, 
+          tempOverlappedEventAttendees);
+      // Copy optional attendees into a modifiable collection
+      optionalOverlappedAttendees = 
+        copyToModifiableCollection(
+          optionalOverlappedAttendees, 
+          tempOverlappedEventAttendees);
```

Comment:
I think I wasn't able to do that because addAll() throws UnsupportedOperationException. The objects I am working with here were declared as an unmodifiableCollection and unmodifiableSet in the pre-existing code of MeetingRequest.java and Event.java. Therefore, in order to filter the lists for relevant attendees, I had to manually do it using my own function.

Reply:
Then I think the option would be to just make a copy so you could do:

`Set<String> overlappedEventAttendees = new HashSet<String>(tempOverlappedEventAttendees);`

Reply:
Oh, wow haha that's a very simple fix compared to how I did it, thanks!